### ISSUE TITLE
check_ncpa.py URL encoding doesnt work

### ISSUE THREAD

**Issue by ccztux** [NagiosEnterprises/ncpa#228]
Hi!

I have tested the actual check_ncpa.py from master branch and in this version, the URL encoding doesnt work:
```
Connecting to: https://my_host:5693/api/interface/Local Area Connection/bytes_recv/?token=my_token&warning=100&critical=120&delta=True&units=M&check=1
The stack trace:Traceback (most recent call last):
  File "./check_ncpa.py", line 277, in main
    info_json = get_json(options)
  File "./check_ncpa.py", line 228, in get_json
    ret = urlopen(url)
  File "/usr/lib64/python2.6/urllib2.py", line 126, in urlopen
    return _opener.open(url, data, timeout)
  File "/usr/lib64/python2.6/urllib2.py", line 397, in open
    response = meth(req, response)
  File "/usr/lib64/python2.6/urllib2.py", line 510, in http_response
    'http', request, response, code, msg, hdrs)
  File "/usr/lib64/python2.6/urllib2.py", line 435, in error
    return self._call_chain(*args)
  File "/usr/lib64/python2.6/urllib2.py", line 369, in _call_chain
    result = func(*args)
  File "/usr/lib64/python2.6/urllib2.py", line 518, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
HTTPError: HTTP Error 400: Bad Request
```

If i enter the URL encoded it works:
```
Connecting to: https://my_host:5693/api/interface/Local%20Area%20Connection/bytes_recv/?token=my_token&warning=100&critical=120&delta=True&units=M&check=1
File returned contained:
{
  "value": {
    "returncode": 0,
    "stdout": "OK: bytes_recv was 0Mb/sec | 'bytes_recv_0'=0Mb;100;120;"
  }
}
The stack trace:Traceback (most recent call last):
  File "./check_ncpa.py", line 277, in main
    info_json = get_json(options)
  File "./check_ncpa.py", line 233, in get_json
    arr = json.load(ret)
  File "/usr/lib64/python2.6/json/__init__.py", line 267, in load
    parse_constant=parse_constant, **kw)
  File "/usr/lib64/python2.6/json/__init__.py", line 307, in loads
    return _default_decoder.decode(s)
  File "/usr/lib64/python2.6/json/decoder.py", line 319, in decode
    obj, end = self.raw_decode(s, idx=_w(s, 0).end())
  File "/usr/lib64/python2.6/json/decoder.py", line 338, in raw_decode
    raise ValueError("No JSON object could be decoded")
ValueError: No JSON object could be decoded
```

**Comment by ccztux** [comment]
I have fixed this, but i dont know, if this is a proper solution:
```
[root@localhost ~]# diff old_check_ncpa.py check_ncpa.py
162,163c162,163
<         api_address = 'https://%s:%d/api/%s/%s' % (hostname, port, metric,
<                                                    arguments)
---
>
>         api_address = 'https://%s:%d/api/%s/%s' % (hostname, port, urlquote(metric), arguments)

```

**Comment by jhomann** [comment]
Fixed in 5160983af80485088f5b348814fcecda4c9c7219, you were basically right with your solution actually. The metrics portion was never encoded. I just placed the urlquote a couple lines up.
